yb-voyager/cmd/conflictDetectionCache.go

```diff
@@ -107,10 +159,12 @@ func (c *ConflictDetectionCache) eventsConfict(event1, event2 *tgtdb.Event) bool
 	}
 	uniqueKeyColumns := c.tableToUniqueKeyColumns[maybeQualifiedName]
 	for _, column := range uniqueKeyColumns {
-		// if the unique key column value is same, then we have a conflict
-		if *event1.Fields[column] == *event2.Fields[column] {
+		if event1.BeforeFields[column] == nil || event2.Fields[column] == nil {
+			return false
+		}
+		if utils.CompareWithoutQuotes(*event1.BeforeFields[column], *event2.Fields[column]) {
```

Comment:
This is much more clear now. you will be comparing the "before" value of the cachedEvent to the "after" value of the incomingEvent to see if they match 👍